esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,527 @@
+        module = importlib.import_module(module_name)
+        logger.info("CMORizing dataset %s using Python script %s", dataset,
+                    module.__file__)
+        cmor_cfg = read_cmor_config(dataset)
+        module.cmorization(in_dir, out_dir, cmor_cfg, self.config, start, end)
+        logger.info('CMORization of dataset %s finished!')
+        return True
+
+
+class DataCommand():
+    """Download and format data to use with ESMValTool."""
+    def __init__(self):
+        datasets_file = os.path.join(os.path.dirname(__file__), 'datasets.yml')
+        with open(datasets_file) as data:
+            self._info = yaml.safe_load(data)
+        self.formatter = Formatter(self._info)
+
+    def list(self):
+        """List all supported datasets."""
+        print()
+        print(f'| {"Dataset name":30} | Tier | Auto-download | Last access |')
+        print('-' * 71)
+        for dataset, dataset_info in self._info['datasets'].items():
+            date = datetime.datetime.strptime(str(dataset_info['last_access']),
+                                              "%Y-%m-%d")
+            print(f"| {dataset:30} | {dataset_info['tier']:4} "
+                  f"| {self.formatter.has_downloader(dataset):13} "
+                  f"|  {date.strftime('%Y-%m-%d')} |")
+        print('-' * 71)
+
+    def info(self, dataset):
+        """Show detailed info about a specific dataset.
+
+        Parameters
+        ----------
+        dataset : str
+            dataset to show
+        """
+        dataset_info = self._info['datasets'][dataset]
+        print(dataset)
+        print()
+        print(f"Tier: {dataset_info['tier']}")
+        print(f"Source: {dataset_info['source']}")
+        print(f"Automatic download: {self.formatter.has_downloader(dataset)}")
+        print("")
+        print(dataset_info['info'])
+
+    def download(self,
+                 datasets,
+                 config_file=None,
+                 start=None,
+                 end=None,
+                 overwrite=False,
+                 **kwargs):
+        """Download datasets.
+
+        Parameters
+        ----------
+        datasets : list(str), optional
+            List of datasets to format, by default None
+        config_file : str, optional
+            Path to ESMValTool's config user file, by default None
+        start : str, optional
+            Start of the interval to process, by default None. Valid formats
+            are YYYY, YYYYMM and YYYYMMDD.
+        end : str, optional
+            End of the interval to process, by default None. Valid formats
+            are YYYY, YYYYMM and YYYYMMDD.
+        overwrite : bool, optional
+            If true, download already present data again
+        """
+        start = self._parse_date(start)
+        end = self._parse_date(end)
+
+        self.formatter.start('download', datasets, config_file, kwargs)
+        self.formatter.download(start, end, overwrite)
+
```

Comment:
Is there any particular reason for using `start` and `end` in this class instead of `start_date` and `end_date` elsewhere? This is rather confusing because all downloader scripts use the latter. For example, the command `esmvaltool data download --start_date=1999 --end_date=2000 CDS-SATTELITE-SOIL-MOISTURE` downloads the dataset for the whole default time range hard-coded in that download script with `start_date=1978` and `end_date=2020` and not [1999-2000]. Could we use `start_date` and `end_date` for the commands as well?

Reply:
It was just to make it shorter for the CLI. That can be changed if you think consistency is better

Reply:
I tend to think consistency is better in this case but I don't have a strong opinion on this. Users would tend to prefer shorter commands while contributors to cmorizers may get confused while developing downloading scripts on their own.